addons/website_quote/static/src/js/website_quotation.js

```diff
@@ -17,26 +17,97 @@ if(!$('.o_website_quote').length) {
         },
         onClick: function(ev){
             ev.preventDefault();
-            var self = this;
             var href = this.$el.attr("href");
-            var order_id = href.match(/order_id=([0-9]+)/);
-            var line_id = href.match(/update_line\/([0-9]+)/);
-            var token = href.match(/token=(.*)/);
-            ajax.jsonRpc("/quote/update_line", 'call', {
-                'line_id': line_id[1],
-                'order_id': parseInt(order_id[1]),
-                'token': token[1],
-                'remove': self.$el.is('[href*="remove"]'),
-                'unlink': self.$el.is('[href*="unlink"]')
-            }).then(function (data) {
-                if(!data){
-                    location.reload();
-                }
-                self.$el.parents('.input-group:first').find('.js_quantity').val(data[0]);
-                $('[data-id="total_amount"]>span').html(data[1]);
-            });
+            var order_id = href.match(/order_id=([0-9]+)/)[1];
+            var line_id = href.match(/update_line(_dict)?\/([0-9]+)/)[2];
+            var token = href.match(/token=([\w\d-]*)/)[1];
+
+            var callParams = {
+                'line_id': line_id,
```

Comment:
Since we have a diff on the line -> could add `parseInt` to be extra safe and consistent with `order_id`.